### PR TITLE
fix(ci): check command out for error when git fails

### PR DESCRIPTION
**Pull request by BolajiOlajide**
Closes [#1110](https://github.com/sourcegraph/devx-support/issues/1110)
Closes DINF-96

We don't print the stdErr when a command fails … in particular when git fails. Therefore we see very little in the panic of what went wrong.

Explanation:
 > There's a weird behavior that occurs where an error isn't accessible in the err variable
// from a *Cmd executing a git command after calling CombinedOutput().
// This occurs due to how Git handles errors and how the exec package in Go interprets the command's output.
// Git often writes error messages to stderr, but it might still exit with a status code of 0 (indicating success).
// In this case, CombinedOutput() won't return an error, but the error message will be in the out variable.

## Test plan

Manual testing

```go
func main() {
	ctx := context.Background()
	cmd := exec.CommandContext(ctx, "git", "rev-parse", "--is-inside-work-tree")
	out, err := handleGitCommandExec(cmd)
	if err != nil {
		// er := errors.Wrap(err, fmt.Sprintf("idsdsd: %s", string(out)))
		panic(err)
	}
	fmt.Println("hello", string(out))
}
```

### CODE REVIEW

**Comment by Strum355** [comment]
Im wondering whether we should do something more generic for this, because this tends to come up anywhere exec.Command is used :smile: imo we should have a generic exec.Command wrapper that includes stderr + the exit code in the returned error always. Thoughts @BolajiOlajide @burmudar ?

**Comment by burmudar** [comment]
> Im wondering whether we should do something more generic for this, because this tends to come up anywhere exec.Command is used 😄 imo we should have a generic exec.Command wrapper that includes stderr + the exit code in the returned error always. Thoughts @BolajiOlajide @burmudar ?

I agree with you @Strum355 - I see a few attempts at this has been made if I look at `sg/internal` like https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/dev/sg/internal/execute/exec.go?L10:6-10:9#tab=references. Then there is also the run package that handles stuff too 🤔.

Maybe a initial wrapper that just does the error stuff and we add a depguard lint check to not allow exec import under `sg/internal` to promote usage of the new package?

**Comment by BolajiOlajide** [comment]
> > Im wondering whether we should do something more generic for this, because this tends to come up anywhere exec.Command is used 😄 imo we should have a generic exec.Command wrapper that includes stderr + the exit code in the returned error always. Thoughts @BolajiOlajide @burmudar ?
> 
> I agree with you @Strum355 - I see a few attempts at this has been made if I look at `sg/internal` like [sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/dev/sg/internal/execute/exec.go?L10:6-10:9](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/dev/sg/internal/execute/exec.go?L10:6-10:9#tab=references). Then there is also the run package that handles stuff too 🤔.
> 
> Maybe a initial wrapper that just does the error stuff and we add a depguard lint check to not allow exec import under `sg/internal` to promote usage of the new package?

Funny thing is I created the linked package lol, I totally forgot that. I'll switch to using that. 
Thanks.

**Comment by Strum355** [comment]
note that the linked package doesnt seem to make use of stderr, Id probably just unconditionally include stderr along with exit code in the `error` returned :smile:

**Comment by BolajiOlajide** [comment]
@Strum355 @burmudar This is ready for another round of review. 

I've moved the shared snippet to `internal/execute` so other consumers can make use of it.

File: internal/execute/git.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+// This occurs due to how Git handles errors and how the exec package in Go interprets the command's output.
+// Git often writes error messages to stderr, but it might still exit with a status code of 0 (indicating success).
+// In this case, CombinedOutput() won't return an error, but the error message will be in the out variable.
+func handleGitCommandExec(cmd *exec.Cmd) ([]byte, error) {
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	if err := cmd.Run(); err != nil {
+		maybeErrMessage := strings.Trim(stderr.String(), "\n")
```
[the remark this answers is not in the document]
Comment (BolajiOlajide) [reply internal/execute/git.go:41]: Updated. One more final round of review, maybe?

**Review by burmudar — APPROVED** [review]

File: internal/execute/git.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (burmudar) [reply internal/execute/git.go:41]: LGTM!

**Pull request merged**